src/com/sap/piper/integration/TransportManagementService.groovy

```diff
@@ -68,20 +68,39 @@ class TransportManagementService implements Serializable {
 
         def proxy = config.proxy ? config.proxy : script.env.HTTP_PROXY
 
-        script.sh """#!/bin/sh -e
-                curl ${proxy ? '--proxy ' + proxy + ' ' : ''} -H 'Authorization: Bearer ${token}' -F 'file=@${file}' -F 'namedUser=${namedUser}' -o responseFileUpload.txt  --fail '${url}/v2/files/upload'
+        def responseFileUpload = 'responseFileUpload.txt'
+
+        def responseContent
+
+        try {
+            script.sh """#!/bin/sh -e
+                curl ${proxy ? '--proxy ' + proxy + ' ' : ''} -H 'Authorization: Bearer ${token}' -F 'file=@${file}' -F 'namedUser=${namedUser}' -o ${responseFileUpload}  --fail '${url}/v2/files/upload'
             """
 
-        def responseContent = script.readFile("responseFileUpload.txt")
+            echo("File upload successful.")
 
-        if (config.verbose) {
-            echo("${responseContent}")
-        }
+        } catch (Exception e) {
+            String message = 'Exception caught during file upload.'
+            if (config.verbose) {
```

Comment:
Why `!config.verbose`? This means we reveal more details in case we are not verbose ...
We can think about dropping the verbose check in total and provide the details anyway in case we get an exception.

Reply:
The code says: if we are in verbose mode, add to the error message "Consider re-running in verbose mode in order to get more details." Why one should consider re-running in verbose mode, if he/she is already in verbose mode and already should have all the details in place? Or do I miss something obvious?

Reply:
> We can think about dropping the verbose check in total and provide the details anyway in case we get an exception.

Yes, good point.

Reply:
That discussion here is obsolete when we have the other approch with `curl --write-out` instead of `curl --fail`.